### PR TITLE
Add notes about ES and WP min version bump

### PR DESCRIPTION
**Pull request by felipeelia**
Updates info about min. versions for ES and WP

Related to #2431

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,15 @@ All notable changes to this project will be documented in this file, per [the Ke
...
+**Note that the upcoming ElasticPress 4.0.0 release will remove built assets from the `develop` branch, will replace `master` with `trunk`, ~~will build a stable release version including built assets into a `stable` branch,~~ will add a zip with the plugin and its built assets in the GitHub release page, and will include a build script should you want to build assets from a branch.**  As such, please plan to update any references you have from `master` to ~~either `stable` or~~ `trunk` or to GitHub releases depending on whether you require built assets or not.
 
-Official PHP support is currently 5.6+. Minimum PHP version for ElasticPress 4.0.0 will be 7.0+.
+Minimum supported versions:
+
+||Current (3.6.4)|4.0.0|
+|---|:---:|:---:|
+|Elasticsearch|5.0+|5.2+|
+|WordPress|3.7.1+|5.6+|
```
Comment (jeffpaul) [inline CHANGELOG.md:16]: Such a bonkers jump, but I love it (let's get more folks on current versions).

File: README.md [hunk 2]
```diff
@@ -4,13 +4,21 @@
 
 [![Support Level](https://img.shields.io/badge/support-active-green.svg)](#support-level) [![Tests Status](https://github.com/10up/ElasticPress/actions/workflows/test.yml/badge.svg?branch=develop)](https://github.com/10up/ElasticPress) [![Release Version](https://img.shields.io/github/release/10up/ElasticPress.svg)](https://github.com/10up/ElasticPress/releases/latest) ![WordPress tested up to version](https://img.shields.io/badge/WordPress-v5.8%20tested-success.svg) [![MIT License](https://img.shields.io/github/license/10up/ElasticPress.svg)](https://github.com/10up/ElasticPress/blob/develop/LICENSE.md)
 
+* Check out the [ElasticPress Docs](http://10up.github.io/ElasticPress/)
+
 **Please note:** master is the stable branch.
```
Comment (jeffpaul) [inline README.md:9]: May want to add on to this line with something like ", but that will be changing with the 4.0.0 release (read more on that below)."

**Review by jeffpaul — APPROVED** [review]
Left one aside note and a potential edit, but generally good to go... thanks!

**Review by brandwaffle — COMMENTED** [review]
looks good!

**Pull request merged**